src/lib.rs

```diff
@@ -214,22 +370,103 @@ impl ErrorPoint {
     }
 }
 
+
+/// represent a key-value pair
+#[derive(Serialize, Debug, Clone)]
+pub struct KvPair {
+    key: &'static str,
+    value: ThrowContextValues,
+}
+
+impl KvPair {
+    fn new(key: &'static str, value: ThrowContextValues) -> KvPair {
+        KvPair { key, value }
+    }
+    ///key
```

Comment:
I can fill these in post-merge, but just want to note that we'll want more descriptive method documentation.